### PR TITLE
HEH-UI-80: Added role guard for moderator and admin pages

### PR DESCRIPTION
**Pull request by akhaulin**

### CODE REVIEW

File: heh-ui/src/app/components/header/header.component.ts [hunk 1]
```diff
@@ -20,7 +21,20 @@ export class HeaderComponent {
...
   };
 
-  constructor(private router: Router) { }
+  constructor(private router: Router,
+              private roleService: RoleService) {
+    this.tabs = [];
+  }
+
+  getTabs(): any {
+    if (this.roleService.getRoles().administrator) {
```
Comment (MaksimViarenich) [inline heh-ui/src/app/components/header/header.component.ts:30]: `const role = this.roleService.getRoles();`

```
switch(true) {
    case role.administrator:
    case role.moderator:
...
}
```

File: heh-ui/src/app/role-guard/role.guard.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  constructor(private roleService: RoleService,
+              private toaster: ToasterService,
+              private router: Router) {
+  }
+  canActivate(): any {
+    if (this.roleService.getRoles().administrator) {
+      return true;
+    } else {
+      this.router.navigate(['/discounts']);
+      this.toaster.open('You don\`t have administrator permissions');
```
Comment (MaksimViarenich) [inline heh-ui/src/app/role-guard/role.guard.ts:20]: Remove

File: heh-ui/src/app/services/role-service/role.service.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  constructor() {
+    this.roles = {
+      administrator: false,
+      moderator: false,
+      employee: false,
+    };
+  }
+
+  getRoles(): any {
+    const token = localStorage.getItem('isAuth');
```
Comment (MaksimViarenich) [inline heh-ui/src/app/services/role-service/role.service.ts:20]: Add one empty line after `const`

**Review by MaksimViarenich — CHANGES_REQUESTED** [review]

File: heh-ui/src/app/components/header/header.component.ts (hunk 1 above)
Comment (akhaulin) [reply heh-ui/src/app/components/header/header.component.ts:30]: Fixed with the last commit

File: heh-ui/src/app/role-guard/role.guard.ts (hunk 2 above)
Comment (akhaulin) [reply heh-ui/src/app/role-guard/role.guard.ts:20]: Fixed with the last commit

File: heh-ui/src/app/components/header/header.component.ts (hunk 1 above)
Comment (akhaulin) [reply heh-ui/src/app/components/header/header.component.ts]: fixed

**Pull request merged**